lightclient/src/background.rs

```diff
@@ -247,7 +274,7 @@ impl<TPlatform: PlatformRef, TChain> BackgroundTask<TPlatform, TChain> {
                             "Cannot send method response to id={id} chain={chain_id:?}",
                         );
                     }
-                } else if let Some((sub_id_sender, sender)) =
+                } else if let Some(subscription_id_state) =
                     self.id_to_subscription.remove(&(id, chain_id))
```

Comment:
Nit (and not related to this PR): It feels more costly than necessary to remove the subscription from the map every time a message comes in and then put it back after (assuming success). I reckon it'd be more efficienct to loko up the subscription, but only remove it in case of an error (which we'd hope would be less common)

Reply:
Ah I see; this is what you do for actual subscription messages below, and this is just the inital response to the subscription if I am reading right, so it's only a one-time thing per subscription